src/service/extensions-impl.js

```diff
@@ -314,14 +314,16 @@ export class Extensions {
           try {
             factory(/** @type {!ShadowRoot} */ (ampdoc.getRootNode()));
           } catch (e) {
-            rethrowAsync('ShadowRoot factory failed: ', e, extensionId);
+            rethrowAsync(dev().createError('ShadowRoot factory failed: ',
```

Comment:
Ditto: we are re-marking a potential "user" error as "dev" error here. This is obviously not very likely during factory installation, but then why risk it?